### ISSUE TITLE
"Found argument '--alloc' which wasn't expected" in loader build.rs

### ISSUE THREAD

**Issue by HazeSystem** [ayrtonm/psx-sdk-rs#21]
Got the following error:

```
error: failed to run custom build command for `loader v0.1.0 (C:\Users\renhe\Documents\GitHub\psx-sdk-rs\examples\loader)`

Caused by:
  process didn't exit successfully: `C:\Users\renhe\Documents\GitHub\psx-sdk-rs\examples\loader\target\release\build\loader-63330a19a9493fb1\build-script-build` (exit code: 2)
  --- stdout
  cargo:rerun-if-changed=C:\Users\renhe\Documents\GitHub\psx-sdk-rs\examples\ferris

  --- stderr
  error: Found argument '--alloc' which wasn't expected, or isn't valid in this context

        If you tried to supply `--alloc` as a value rather than a flag, use `-- --alloc`

  USAGE:
      cargo-psx.exe --lto <PSX> <build|check|run|test>
```

Seems it's because of the `--alloc` flag being removed in [this](https://github.com/ayrtonm/psx-sdk-rs/commit/7d4c75949ded65e6a085a80c76326f31afff0b86) commit.

I tried removing the flag, and instead get these errors:

```
error[E0425]: cannot find value `MAIN_RAM` in this scope
  --> src\main.rs:29:34
   |
29 |         let load_addr = 524288 + MAIN_RAM as usize + BIOS_LEN - 2048;
   |                                  ^^^^^^^^ not found in this scope

error[E0308]: mismatched types
   --> src\main.rs:39:19
    |
39  |         file.read(exe).expect("Could not read PROG2.EXE");
    |              ---- ^^^ expected `u8`, found `u32`
    |              |
    |              arguments to this method are incorrect
    |
    = note: expected mutable reference `&mut [u8]`
               found mutable reference `&mut [u32]`
note: associated function defined here
   --> C:\Users\renhe\Documents\GitHub\psx-sdk-rs\psx\src\sys\fs.rs:339:12
    |
339 |     pub fn read(&self, dst: &mut [u8]) -> Result<usize, Error<T>> {
    |            ^^^^
```

**Comment by ayrtonm** [comment]
Sorry about this one! I'll add a README and keep cargo-psx args more stable now that at least one other person is using it :). I need to change the way crate features are specified in cargo-psx because it's a bit ugly rn, so expect a bit of churn in the next few days...

If you want to build the demo you're right about needing to remove `--alloc`. Then just change `MAIN_RAM` to [`KSEG0`](http://problemkaputt.de/psx-spx.htm#memorymap) and change the slice on line 35 to a u8 slice (and fix its size).

**Comment by ayrtonm** [comment]
I fixed the build in #26, but I didn't see the ferris demo getting loaded when I ran it in duckstation. I'm not sure if it's an issue with the demo or my duckstation config though.
